scripts/main/lychee.js

```diff
@@ -383,23 +561,42 @@ lychee.parsePublicInitializationData = function (data) {
 	lychee.nsfw_blur = data.config.nsfw_blur === "1";
 	lychee.nsfw_warning = data.config.nsfw_warning === "1";
 
-	lychee.header_auto_hide = data.config_device.header_auto_hide;
-	lychee.active_focus_on_page_load = data.config_device.active_focus_on_page_load;
-	lychee.enable_button_visibility = data.config_device.enable_button_visibility;
-	lychee.enable_button_share = data.config_device.enable_button_share;
-	lychee.enable_button_archive = data.config_device.enable_button_archive;
-	lychee.enable_button_move = data.config_device.enable_button_move;
-	lychee.enable_button_trash = data.config_device.enable_button_trash;
-	lychee.enable_button_fullscreen = data.config_device.enable_button_fullscreen;
-	lychee.enable_button_download = data.config_device.enable_button_download;
-	lychee.enable_button_add = data.config_device.enable_button_add;
-	lychee.enable_button_more = data.config_device.enable_button_more;
-	lychee.enable_button_rotate = data.config_device.enable_button_rotate;
-	lychee.enable_close_tab_on_esc = data.config_device.enable_close_tab_on_esc;
-	lychee.enable_tabindex = data.config_device.enable_tabindex;
-	lychee.enable_contextmenu_header = data.config_device.enable_contextmenu_header;
-	lychee.hide_content_during_imgview = data.config_device.hide_content_during_imgview;
-	lychee.device_type = data.config_device.device_type || "desktop"; // we set default as Desktop
+	lychee.sm_facebook_url = data.config.sm_facebook_url;
+	lychee.sm_flickr_url = data.config.sm_flickr_url;
+	lychee.sm_instagram_url = data.config.sm_instagram_url;
+	lychee.sm_twitter_url = data.config.sm_twitter_url;
+	lychee.sm_youtube_url = data.config.sm_youtube_url;
+
+	lychee.site_title = data.config.site_title;
+	lychee.site_owner = data.config.site_owner;
+	lychee.site_copyright_begin = data.config.site_copyright_begin;
+	lychee.site_copyright_end = data.config.site_copyright_end;
+
+	lychee.footer_show_social_media = data.config.footer_show_social_media === "1";
+	lychee.footer_show_copyright = data.config.footer_show_copyright === "1";
+	lychee.footer_additional_text = data.config.footer_additional_text;
+
+	lychee.mod_frame_enabled = data.config.mod_frame_enabled === "1";
+	lychee.mod_frame_refresh = Number.parseInt(data.config.mod_frame_refresh, 10) || 30;
```

Comment:
Totally agreed from a software architecture perspective. Actually, this whole code should not even need to exist at all.

But there was a reason why I kept it like that. I wanted to keep this PR "small" in the sense that I only wanted to merge the previously distinct modes "view", "frame" and "gallery" into one and otherwise change as little as possible and only "move" code around but not actually change anything.

I hoped that this approach might speed up the review process. (I would like to here @kami4 opinion about that.) As I said, this PR is not supposed to be the last one or even be "good" by any standards. The frontend is complete spaghetti code.

My final intention is to completely clean up the architecture of the frontend eventually (hopefully the week after the next, if time permits). Then, I would like to remove strange JS objects like `build` or `visible` which do not make any sense to me, but only have objects like `ViewManager`, `AbstractView`, `MapView`, `AlbumView`, `PhotoView` and `AlbumController`, `PhotoController`, etc. For example `visible.photo` would then be moved to `PhotoView.isVisible`.

However, I feared if I would start with the step above as the first step for the "gallery" mode only, I might have accidentally break the "view" and the "frame" mode, because they are the unbeloved children. Hence, I decided to take the following approach:

 1. Preparation 1:
      - Merge "gallery", "frame" and "view" (this PR)
      - Move code around as necessary and throw out duplicates, but don't change anything on the fundamental level
 3. Preparation 2: Revamp the box model of the frontend (see https://github.com/LycheeOrg/Lychee-front/pull/335)
 4. Clean up the architecture step-by-step and many individual PRs

Yes, I could have done everything in one "big bang", but then I feared that the PR would become huge and reviews would take forever.

Reply:
I would like to keep this conversation unresolved, until @kamil4 gave his opinion.

Reply:
I'm sort-of wondering myself if many small PRs vs one big one improve the overall throughput, but for sure they cut down on latency. I'm generally in favor of smaller PRs.